### ISSUE TITLE
Bug:  Invalid character in header content ["Content-Disposition"]

### ISSUE THREAD

**Issue by chrisVillanueva** [Moorad/youtubeDownloader#25]
Hello.

I ran into an issue when testing a download.  I encountered this server error:

```
Server Works !!! At port 4000
TypeError [ERR_INVALID_CHAR]: Invalid character in header content ["Content-Disposition"]
    at ServerResponse.setHeader (_http_outgoing.js:473:3)
```

Any ideas on how to resolve this?

Thanks!

**Comment by Moorad** [comment]
Have you modified the code in any way? if you have can you show me that?

Be careful with ``res.header('Content-Disposition', `attachment; filename="${title}.mp3"`);``, `` `attachment; filename="${title}.mp3"` `` is wrapped around with ticks `` ` `` not `'`

**Comment by chrisVillanueva** [comment]
Hi @Moorad 

No, I have not modified the code.  I pulled `master` and started server, then the client html.

**Comment by Moorad** [comment]
ok give me one min

**Comment by Moorad** [comment]
Alright I messed around a little bit and I was able to replicate the same error
![image](https://user-images.githubusercontent.com/34781684/84703154-6d147580-af60-11ea-9d61-4826b864396d.png)
The problem seems to be related to the title of the video. On line 23 where you have ``res.header('Content-Disposition', `attachment; filename="${title}.mp3"`);`` the name of the file is set to the title of the video. However, the filename cannot contain a character outside the ASCII range (for example an emoji) so if the video title contains a character outside the ASCII range it throws a `TypeError`.

 This might look weird but if you add `title = title.replace(/[^\x00-\x7F]/g, "");` (which removes anything non ASCII) above ``res.header('Content-Disposition', `attachment; filename="${title}.mp3"`);`` so the code looks something like
```js
...
	title = info.player_response.videoDetails.title;
});

title = title.replace(/[^\x00-\x7F]/g, "");
res.header('Content-Disposition', `attachment; filename="${title}.mp3"`);
ytdl(url, {
	format: 'mp3',
	filter: 'audioonly',
...
```
it should do the trick
